src/charm.py

```diff
@@ -257,6 +278,8 @@ def _get_all_access_rules_repositories(self) -> Optional[List]:
         if cm_access_rules := self.access_rules_configmap.get():
             for key in cm_access_rules.keys():
                 repositories.append(f"{self._access_rules_dir_path}/{key}")
+        if admin_ui_access_rules_file := self.info_provider.get_requirer_info():
+            repositories.append(admin_ui_access_rules_file)
```

Comment:
What is this for? Are we adding an access rule for the admin UI?

Reply:
These rules are not protecting Admin UI - in that case we'd use `auth-proxy` indeed.
This file contains custom access rules defined **in** Admin UI, outside of the `auth-proxy` relation.

Reply:
Isn't the `configMap` going to be used for this?

Reply:
Ok, my bad - no need to call `self.info_provider.get_requirer_info()` since the configmap will return the admin ui file as well. I corrected it.
There is one issue related to this: the config update somehow needs to be triggered when Admin UI adds the rule, otherwise it may end up not listed in the section. For this we could either rely on `update-status` or `oathkeeper-info-relation-changed`. If we go with the 1st approach I could drop the requirer data. We'd also cover [this](https://github.com/canonical/oathkeeper-operator/pull/56#discussion_r1486062458) case. Does that sound good?

Reply:
As discussed yesterday, I added a check if the admin ui access rule file exists when the relation joins. If not, oathkeeper will create a key with empty value to ensure the file will be listed in oathkeeper config.